src/UI/SignTransactionsModals/index.tsx

```diff
@@ -71,32 +72,71 @@ export const SignTransactionsModals = ({
     verifyReceiverScam
   };
 
-  if (signError || hasTransactions) {
+  const ConfirmScreens = {
+    Ledger: CustomConfirmScreens?.Ledger ?? SignWithLedgerModal,
+    WalletConnect:
+      CustomConfirmScreens?.WalletConnect ?? SignWithWalletConnectModal,
+    Extension: CustomConfirmScreens?.Extension ?? SignWithExtensionModal,
+    Extra: CustomConfirmScreens?.Extra ?? null
+  };
+
+  const renderTransactionStatusToast = () => {
```

Comment:
I think we can turn this login into a single <TransactionStatusToast which gets different props, depending on the type of the flow